### PR TITLE
feat: add/configure dotenv for elixir

### PR DESCRIPTION
**Pull request by malssid**
You can access environment variables with `Dotenv.get("<ENV_VARIABLE>")`

### CODE REVIEW

File: server-phoenix/lib/helios/application.ex [hunk 1]
```diff
@@ -7,6 +7,11 @@ defmodule Helios.Application do
 
   @impl true
   def start(_type, _args) do
+    unless Mix.env == :prod do
+      Dotenv.load
```
Comment (cbarber) [inline server-phoenix/lib/helios/application.ex:11]: Does this find the `.env.local` file in the parent directory?

File: server-phoenix/mix.exs [hunk 2]
```diff
@@ -50,7 +50,8 @@ defmodule Helios.MixProject do
       {:jason, "~> 1.2"},
       {:plug_cowboy, "~> 2.5"},
       {:absinthe, "~> 1.7"},
-      {:absinthe_plug, "~> 1.5"}
+      {:absinthe_plug, "~> 1.5"},
+      {:dotenv, "~> 3.0.0", only: [:dev, :test]}
```
Comment (cbarber) [inline server-phoenix/mix.exs:54]: I'm unclear how this works in production if we have something like `Dotenv.get("<ENV_VARIABLE>")` in code.

Comment (malssid) [reply server-phoenix/mix.exs:54]: I can add the library for all environments by leaving out the "only" option. Are these env variables loaded into heroku directly for production? (through dashboard settings or the cli)

Comment (cbarber) [reply server-phoenix/mix.exs:54]: They are. My question is how will `Dotenv.get` function in the code in production if its only available for `:dev` and `:test`?

File: server-phoenix/lib/helios/application.ex (hunk 1 above)
Comment (cbarber) [reply server-phoenix/lib/helios/application.ex:11]: I'm getting this locally:
```
iex(1)> Dotenv.get("WEATHER_API_KEY")
nil
```

Comment (malssid) [reply server-phoenix/lib/helios/application.ex:11]: That library might only work for .env files, not .env.local

File: server-phoenix/mix.exs (hunk 2 above)
Comment (malssid) [reply server-phoenix/mix.exs:54]: I can remove the "only" option like this:
`{:dotenv, "~> 3.0.0"}`
so then it'll be available in all environments.

**Comment by chewnoill** [comment]
LGTM, I think this is the right thing to do. There is some language/documentation problems though. Not clear if these have been addressed based on PR description/discussion

* ENV should reference `System.get_env("<ENV_VARIABLE>")` and it should be the same way for dev/test/prod. It doesn't make sense to reference them through `Dotenv.get` because this is not loaded in prod.

* If we need to do something with `.env.local` (`cp .env.local .env`) before we can run we should update the readme so that is clear

**Comment by malssid** [comment]
Are you suggesting we use `System.put_env(key, value)` to load them in before reading them with `System.get_env(key)`?

**Comment by chewnoill** [comment]
Yes, I expect that was the whole point of using [Dotenv.load](https://github.com/avdi/dotenv_elixir/blob/master/lib/dotenv.ex#L112)

**Comment by malssid** [comment]
ok yeah, I just tested `System.get_env("WEATHER_API_KEY")` and it works. I'll also update the readme so people know to copy the sample .env.local to a .env.

**Comment by malssid** [comment]
Never mind. It looks like the readme already tells them to copy .env.local.sample to a .env.

**Comment by cbarber** [comment]
* Lets rename `.env.local.sample` to `.env.sample` as well
* webpack.config.js has a hardcoded path to `.env.local` that will need to be updated

File: README.md [hunk 3]
```diff
@@ -34,7 +34,8 @@ Prequisites:
 - `gem install foreman`
 - `bundle install`
 - `yarn install`
-- `cp .env.local.sample .env` -- fill in env vars where applicable
+- `cp .env.sample .env` -- dotenv loads variables from a **.env** in your project's root directory. Fill in env vars where applicable
+- `cp .env.local .env` -- for people with a .env.local
```
Comment (chewnoill) [inline README.md:38]: This seems confusing. Do we need both of these? Can we consolidate them into one file?

Comment (malssid) [reply README.md:38]: Yeah, my bad. Let's just keep the first line that tells them to copy the contents from .env.sample to a .env

**Review by cbarber — APPROVED** [review]
Nice work! :ship: :rocket:

**Pull request merged**